backends/p4fmt/attach.cpp

```diff
@@ -0,0 +1,89 @@
+#include "attach.h"
+
+#include "frontends/common/parser_options.h"
+#include "lib/source_file.h"
+
+namespace P4::P4Fmt {
+
+Attach::Attach() = default;
+Attach::~Attach() = default;
+
+void Attach::addPrefixComments(NodeId node, const Util::Comment *prefix) {
+    commentsMap[node].prefix.push_back(prefix);
+}
+
+void Attach::addSuffixComments(NodeId node, const Util::Comment *suffix) {
+    commentsMap[node].suffix.push_back(suffix);
+}
+
+bool Attach::isSystemFile(const std::filesystem::path &file) {
```

Comment:
Unrelated to your actual PR but this function should probably reside in `parser_options.h`. I will add this as a TODO for myself.

Reply:
@snapdgn A helper function that should be able to do this was merged here: https://github.com/p4lang/p4c/pull/4888